engine/classes/Elgg/Di/ServiceProvider.php

```diff
@@ -282,7 +282,18 @@ public function __construct(Config $config) {
 		$this->setClassName('input', \Elgg\Http\Input::class);
 
 		$this->setFactory('imageService', function(ServiceProvider $c) {
-			$imagine = new \Imagine\Gd\Imagine();
+			switch ($c->config->image_processor) {
+				case 'imagick':
+					if (extension_loaded('imagick')) {
+						$imagine = new \Imagine\Imagick\Imagine();
+						break;
+					}
+				default:
```

Comment:
If it's set and not supported, we should complain loudly.

Reply:
We could as well default to imagick if it's supported. Not sure if GD requirements checked anywhere in the installer

Reply:
I wonder if Imagine has a factory that just picks the best available.

Reply:
I don't think there is a definition of "best" to work with here. Imagick seems to be the most performant